nservicebus/scalability-and-ha/scale-out.md

```diff
@@ -0,0 +1,74 @@
+---
+title: Scaling out NServiceBus endpoints
+summary: How to scale out NServiceBus endpoints
+tags:
+- Scale Out
+- Routing
+---
+
+## Version 5 and below
+
+In Version 5 and below NServiceBus scale out capabilities are dependent on the transport being used.
+
+
+### MSMQ
+
+Because of limitations of MSMQ related to remote receive, in order to scale out an MSMQ Version 5 (and below) endpoint users have to use the [distributor](/nservicebus/scalability-and-ha/distributor/). The role of the distributor is to forward incoming messages to a number of workers in order to balance the load. The workers are "invisible" to the outside world because all the outgoing messages contain the distributor's (not the worker's) address in the `reply-to` header. 
+
+The main issue with distributor is the throughput limitation due to the fact that, for each message forwarded to worker, there were additional two messages exchanged between the worker and the distributor.
+
+
+### SQL Server and RabbitMQ
+
+Up to version 5 (inclusive) both SQL Server and RabbitMQ transports scale out by adding more receivers to the same queue, taking advantage of the competing consumers capability built into these transports. All the instances feeding off the queue have same *endpoint name* and *address* so they appear to the outside world as a single instance.
+
+The benefit of this approach is zero configuration. New instances can be added by `xcopy`-ing the deployment folder. The potential downside is that total throughput of the endpoint is capped to the maximum throughput of a single queue in the underlying infrastructure. 
+
+
+### Azure Storage Queues and Azure Service Bus
+
+Up to version 5 (inclusive) both Azure transports behave similarly to other broker transports (SQL Server and RabbitMQ): they scale out by adding receivers to a single queue. An additional feature of Azure transports is ability to use individualized queues (based on the *role instance ID*)
+
+snippet: UniqueQueuePerEndpointInstance
+
+This feature is, however, *not intended* for scaling out as there is no built-in distribution mechanism between that would route messages to the individualized queues.
+
+## Version 6 and above
+
+Version 6 of NServiceBus comes with a unified scalability model which is based on the concept of endpoint instance ID. Each deployment of NServiceBus can (but does not have to) be assigned an *instance ID*. Instance IDs can be shared between several deployments. 
```

Comment:
Overriding instance ID for when hosted in the cloud service makes very little sense. What would be the case for that? Also, thinking about the monitoring (perhaps I'm not entirely understanding this) wouldn't that be an issue if two instances report the same ID?